### PR TITLE
Feature/add placeholder for javadocs

### PR DESCRIPTION
**Pull request by albertshau**
adding a placeholder for javadocs so that the table of contents will include a link to the javadocs page.  Javadocs are then built separately and the javadocs directory is replaced manually.

### CODE REVIEW

File: docs/docs/source/overview/overview-links.rst [hunk 1]
```diff
@@ -16,5 +16,4 @@
 :doc:`Features <features>` |
 :doc:`Architecture <architecture>` |
 :doc:`Design Goals <design-goals>` |
-:doc:`Use Cases <use-cases>` |
-:doc:`System Requirements <system-requirements>`
```
Comment (wolf31o2) [inline docs/docs/source/overview/overview-links.rst:20]: Why is this removed?

File: docs/docs/source/rest/rest-links.rst [hunk 2]
```diff
@@ -20,5 +20,4 @@
 :doc:`Services </rest/services>` |
 :doc:`Cluster Template </rest/templates>` |
 :doc:`Importing & Exporting </rest/entities>` |
-:doc:`Clusters </rest/clusters>` |
-:doc:`Tasks </rest/tasks>`
```
Comment (wolf31o2) [inline docs/docs/source/rest/rest-links.rst:24]: Why is Tasks removed?

File: docs/docs/source/overview/overview-links.rst (hunk 1 above)
Comment (albertshau) [inline docs/docs/source/overview/overview-links.rst:20]: this was a warning, link to something that doesn't exist

File: docs/docs/source/rest/rest-links.rst (hunk 2 above)
Comment (albertshau) [inline docs/docs/source/rest/rest-links.rst:24]: this was a warning, link to something that doesn't exist

**Comment by wolf31o2** [comment]
:+1: LGTM, then
